src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/special/SpecialWikiActivityPageObject.java

```diff
@@ -129,15 +127,20 @@ public Boolean doesLastNRecentBlogActivitiesContain(int n, String blogPostConten
 
   /**
    * Gets activities from special page, limited by provided value
-   * @param numberOfActivities
-   * @return
+   * @param numberOfActivities number of activities to fetch (counting from top)
+   * @return list of activities
    */
   public List<Activity> getActivities(int numberOfActivities) {
 
     List<Activity> activityList = new ArrayList<>();
     //numberOfActivitiesToFetch set to max number of possible elements to fetch
-    int numberOfActivitiesToFetch = numberOfActivities > activityWebElementList.size()
-            ? activityWebElementList.size() : numberOfActivities;
+    int numberOfActivitiesToFetch;
+    if (activityWebElementList.size()>0){
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1155)